examples/motoko_examples/quicksort/canisters/rust/src/quicksort.rs

```diff
@@ -0,0 +1,60 @@
+use ic_cdk::export::candid::Int;
+use std::{cmp::Ordering, convert::TryInto};
+
+trait AbsoluteValue {
+    fn abs(&self) -> usize;
+}
+
+impl AbsoluteValue for Int {
+    fn abs(&self) -> usize {
+        let positive_int = if self >= &Int::from(0i8) {
+            self.clone()
+        } else {
+            self.clone() * Int::from(-1i8)
+        };
+        positive_int.0.try_into().unwrap()
+    }
+}
+
+pub fn sort_by<X: Clone + Ord, T: Fn(&X, &X) -> Ordering>(xs: Vec<X>, f: &T) -> Vec<X> {
+    let n = xs.len();
+    if n < 2 {
+        return xs;
+    } else {
+        let mut result = xs.clone();
+        sort_by_helper(&mut result, 0.into(), (n - 1).into(), f);
+        result
+    }
+}
+
+fn sort_by_helper<X: Clone + Ord, T: Fn(&X, &X) -> Ordering>(
+    xs: &mut Vec<X>,
+    l: Int,
+    r: Int,
+    f: &T,
+) -> () {
+    if l < r {
+        let mut i = l.clone();
```

Comment:
Let's just see if we can get rid of all of these clones

Reply:
I don't think that's possible. If these were on the stack then assigning them to multiple variables would be no problem because it would be a cheap copy. But because Int is stored on the heap, setting `i` to `l` would actually be a `move` which means `l` wouldn't be available anymore. Thus, the copy. If I'm mistaken or there's a way to do this I'm happy to work through this with you.